demo/shell.py

```diff
@@ -0,0 +1,67 @@
+#
+# Copyright 2015 The ndb Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import atexit
+import code
+import readline
+import os
+import sys
+
+import ndb
+from ndb import tasklets
+from ndb.google_imports import datastore_pbs
+
+HISTORY_PATH = os.path.expanduser('~/.ndb_shell_history')
+
+def shell():
+
+  if (not os.environ.get('DATASTORE_APP_ID', None)
+      and not os.environ.get('DATASTORE_PROJECT_ID', None)):
+    raise ValueError('Must set either DATASTORE_APP_ID or DATASTORE_PROJECT_ID'
```

Comment:
Not necessary. Lack of one of these variables will cause ndb to fallback to the GAE version and complain about missing the stubs. If the user sets DATASTORE_PROJECT_ID but not DATASTORE_USE_PROJECT_ID_AS_APP_ID it will fail when creating the context with a good error message.